edx_toggles/tests/test_waffle.py

```diff
@@ -0,0 +1,37 @@
+"""
+Unit tests for waffle classes.
+"""
+
+from django.test import TestCase
+
+# TODO import from edx_toggles.toggles once we remove the legacy classes from the exposed API
+from edx_toggles.toggles.internal.waffle.flag import WaffleFlag
+from edx_toggles.toggles.internal.waffle.switch import WaffleSwitch
+
+
+class WaffleFlagTests(TestCase):
+    """
+    WaffleFlag tests.
+    """
+
+    def test_name_validation(self):
+        WaffleFlag("namespaced.name", module_name="module1")
+        self.assertRaises(
+            ValueError, WaffleFlag, "non_namespaced", module_name="module1"
+        )
```

Comment:
1. I thought you needed to use as context?  For example:
```
        with self.assertRaises(
            ValueError, WaffleFlag, "non_namespaced", module_name="module1"
        ):
            WaffleFlag("namespaced.name", module_name="module1")
```
2. This might be passing as-is (if it is) because you are in fact sending it a namespaced name.

Reply:
Actually, I never used `assertRaises` as a context. It is possible to pass a callable and its arguments to assertRaises: https://docs.python.org/3.7/library/unittest.html#unittest.TestCase.assertRaises
I like the context usage better, so I'll use it here.